GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java

```diff
@@ -1377,37 +1377,37 @@ private String sanitize(String s) {
 
                             createCell(row, offset++, header, headerStyle);
 
-                            // check if we need to create columns for all
-                            // options
-                            if (QuestionType.OPTION == q.getType()
-                                    && useQuestionId) {
+                            // check if we need to create columns for all options
+                            if (QuestionType.OPTION == q.getType() && useQuestionId) {
 
                                 // get options for question and create columns
-                                OptionContainerDto ocDto = q
-                                        .getOptionContainerDto();
-                                List<QuestionOptionDto> qoList = ocDto
-                                        .getOptionsList();
-
-                                for (QuestionOptionDto qo : qoList) {
-                                    // create header column
-                                    header = (qo.getCode() != null
-                                            && !qo.getCode().equals("null") && qo
-                                            .getCode().length() > 0) ? qo
-                                            .getCode() + ":" : "";
-                                    createCell(row, offset++, "--OPTION--|"
-                                            + header + qo.getText(),
-                                            headerStyle);
-                                }
-
-                                // add 'other' column if needed
-                                if (q.getAllowOtherFlag()) {
-                                    createCell(row, offset++, "--OTHER--",
-                                            headerStyle);
+                                OptionContainerDto ocDto = q.getOptionContainerDto();
+                                if (ocDto != null) { //used to be legal
```

Comment:
Just as an extra check I would suggest we check that the list has items and not just that it is `null`